lib/puppet-lint/plugins/check_parameter_documentation.rb

```diff
@@ -1,5 +1,8 @@
 PuppetLint.new_check(:parameter_documentation) do
   def check
+
+    ignore_private = PuppetLint.configuration.docs_ignore_private_api.nil? ? true : PuppetLint.configuration.docs_ignore_private_api
```

Comment:
I looked into a cleaner way to add configuration but it looks like there isn't really a good way. Perhaps it can be simplified to
```suggestion
    ignore_private = PuppetLint.configuration.docs_ignore_private_api.nil? || PuppetLint.configuration.docs_ignore_private_api
```

Not sure if it's cleaner. The cleanest way would be to have an API to also add a default.

Perhaps a much better way is to avoid the negative name and name it `docs_check_private_params`. That can default to `false`. Since `nil` and `false` are equivalent in this case, it would solve the problem.

The only downside is that the `unless` condition becomes more complex, but overall I think it's a win. Let me know what you think.

Reply:
I have no opinion on the naming. adjustments have been made :)